### ISSUE TITLE
New test methods are not run until i restart VS

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#36123]
I've been noticing this a while, but haven't gotten around to filing this until now.  For reasons i cannot figure out, the test-runner in VS never wants to run tests i've just added.  For example, i added this test:

![image](https://user-images.githubusercontent.com/4564579/58766842-82c32280-8538-11e9-9707-503797fbc24c.png)

The test then shows up in the test explorer:

![image](https://user-images.githubusercontent.com/4564579/58766849-8f477b00-8538-11e9-9f91-e2edcb68d7ea.png)

But no matter what i do, i cannot get any sort of test run to actually execute it.  Sayng "run test" in the method body doesn't work.   Right clicking on the test in the explorer and trying to run it doesn't work.

The output shows this:

```
[6/2/2019 1:14:14.794 PM Informational] ========== Run finished: 122 tests run (0:00:30.7122483) ==========
[6/2/2019 1:16:12.582 PM Informational] Executing test method: Microsoft.CodeAnalysis.Editor.VisualBasic.UnitTests.UseConditionalExpression.UseConditionalExpressionForReturnTests.TestMissingWhenCrossingPreprocessorDirective
[6/2/2019 1:16:12.837 PM Informational] ---------- Run started ----------
[6/2/2019 1:16:13.935 PM Warning] Test run will use DLL(s) built for framework .NETFramework,Version=v4.5 and platform X86. Following DLL(s) do not match framework/platform settings.
Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.UnitTests.dll is built for Framework 4.7.2 and Platform AnyCPU.
Go to http://go.microsoft.com/fwlink/?LinkID=236877&clcid=0x409 for more details on managing these settings.

[6/2/2019 1:16:14.555 PM Informational] [xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.4.1-pre.build.4059 (32-bit Desktop .NET 4.0.30319.42000)
[6/2/2019 1:16:15.245 PM Informational] [xUnit.net 00:00:00.69]   Discovering: Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.UnitTests
[6/2/2019 1:16:19.553 PM Informational] [xUnit.net 00:00:05.00]   Discovered:  Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.UnitTests
[6/2/2019 1:16:19.638 PM Informational] [xUnit.net 00:00:05.08]   Starting:    Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.UnitTests
[6/2/2019 1:16:23.650 PM Informational] [xUnit.net 00:00:09.09]   Finished:    Microsoft.CodeAnalysis.VisualBasic.EditorFeatures.UnitTests
[6/2/2019 1:16:23.832 PM Informational] [xUnit.net 00:00:00.00] xUnit.net VSTest Adapter v2.4.1-pre.build.4059 (32-bit Universal Windows)
[6/2/2019 1:16:24.121 PM Informational] ========== Run finished: 1 tests run (0:00:11.2767712) ==========
```

And, in the end i see:

![image](https://user-images.githubusercontent.com/4564579/58766866-d59cda00-8538-11e9-969b-0d68d8a8954b.png)

Basically, no matter what i do, *new* test methods just do not get picked up by the *execution* engine, though they are picked up by the UI.  

If i close and open VS, then things are fine and these tests are now run.

It's super annoying (esp. given how long it takes to restart VS).

**Comment by CyrusNajmabadi** [comment]
tagging @tannergooding @kendrahavens

**Comment by kendrahavens** [comment]
Thanks Cyrus! We usually track the unit testing feedback on [developer community](https://developercommunity.visualstudio.com), but maybe @drognanar can take a look here since I don't want to lose the nice formatting.

**Comment by jcouv** [comment]
Tagging @AbhitejJohn for triage/routing.

**Comment by AbhitejJohn** [comment]
From the contents of the Output pane it looks like we've run the test but the result hasn't been updated. I wonder if this is one more case where the source discovered and assembly discovered tests do not match for some reason. Trying this out on roslyn. 
/cc: @shyamnamboodiripad

**Comment by shyamnamboodiripad** [comment]
@CyrusNajmabadi Thanks for reporting this! Could you please include the build number of VS where you are seeing this. Is this 16.2 Preview 1?

**Comment by CyrusNajmabadi** [comment]
I have:

![image](https://user-images.githubusercontent.com/4564579/59081336-63aef280-88a2-11e9-95fc-d8e6487cd210.png)

**Comment by CyrusNajmabadi** [comment]
Note: i also see just plain *busted* behavior like teh following:

Here i'm just trying to run some *existing* tests.  They seem to run.  Nothing else seems to happen, and the UI never updated.  Testing just feels broken in roslyn.

![NoResults](https://user-images.githubusercontent.com/4564579/59133577-55f27f00-892d-11e9-9713-5019a1742df5.gif)

**Comment by tannergooding** [comment]
I've also seen this in other much simpler projects (like ClangSharp, LLVMSharp, and some personal projects).

**Comment by AbhitejJohn** [comment]
@CyrusNajmabadi : I'm guessing you've already tried this out on later previews of 16.2. We've made a bunch of changes/fixes to make the above scenarios more reliable. Please let us know if you are still running into this.
/cc: @shyamnamboodiripad
